File: lib/bit-io/ruby/encode.rb
#!/usr/bin/env ruby

require './bit_io.rb'

if ARGV.length != 2 then
  STDERR.print("Usage: #{$0} INFILE OUTFILE\n")
  exit 2
end

File.open(ARGV[0], 'r') do |input|
  BitIO::BitWriter.open(ARGV[1]) do |output|
    until input.eof? do
      output.writebits(input.readbyte, 7)
    end
  end
end

File: lib/bit-io/python/encode.py
#!/usr/bin/env python

from sys import *
import bit_io

if len(argv) != 3:
    stderr.write('Usage: {} INFILE OUTFILE\n'.format(argv[0]))
    exit(2)

with open(argv[1], 'r') as input, bit_io.BitWriter(argv[2]) as output:
    while True:
        c = input.read(1)
        if not c: break
        output.writebits(ord(c), 7)
